presentation/src/test/java/com/exallium/mvvmcleanexample/presentation/users/UserEditViewModelTest.kt

```diff
@@ -54,33 +63,122 @@ class UserEditViewModelTest {
     fun lastNameChange_inProgress_doesNothing() {
         // GIVEN
         given_routerEmitsResult(ValidateUserLastNameUseCase.Result.InProgress())
+        initTestSubject()
 
         // WHEN
-        when_testSubjectInitialized()
+        testSubject.start()
 
         // THEN
-        assert(testSubject.lastNameError.get().isEmpty())
+        assert(testSubject.lastNameError.isEmpty())
     }
 
     @Test
     fun firstNameChange_inProgress_doesNothing() {
         // GIVEN
         given_routerEmitsResult(ValidateUserFirstNameUseCase.Result.InProgress())
+        initTestSubject()
+
+        // WHEN
+        testSubject.start()
+
+        // THEN
+        assert(testSubject.firstNameError.isEmpty())
+    }
+
+    @Test
+    fun `delegates to notifiableObservable in constructor`() {
+        // GIVEN, WHEN
+        initTestSubject()
+
+        // THEN
+        verify(notifiableObservable).delegateTo(testSubject)
+        verifyNoMoreInteractions(notifiableObservable)
+    }
+
+    @Test
+    fun `subscribes to model on start`() {
+        // GIVEN
+        initTestSubject()
+
+        // WHEN
+        testSubject.start()
+
+        // THEN
+        inOrder(ecoDisposable, savedUserRelay, firstNameRelay, lastNameRelay).apply {
+            verify(savedUserRelay).relay
+            verify(firstNameRelay).relay
+            verify(lastNameRelay).relay
+            verify(ecoDisposable).setValue(any(), any(), any())
+            verifyNoMoreInteractions()
+        }
+    }
+
+    @Test
+    fun `unsubscribes from model on stop`() {
+        // GIVEN
+        initTestSubject()
+
+        // WHEN
+        testSubject.stop()
+
+        // THEN
+        verify(ecoDisposable).setValue(any(), any(), eq(null))
+        verifyNoMoreInteractions(ecoDisposable)
+        verifyZeroInteractions(model, router, savedUserRelay, firstNameRelay, lastNameRelay)
+    }
+
+    @Test
+    fun `notifies firstNameRelay when firstName changed`() {
+        // GIVEN
+        initTestSubject()
+        val expected = "expected"
+
+        // WHEN
+        testSubject.onFirstNameChanged(expected)
+
+        // THEN
+        verify(firstNameRelay).setValue(any(), any(), eq(expected))
+        verifyNoMoreInteractions(firstNameRelay)
+        verifyZeroInteractions(lastNameRelay, savedUserRelay)
+    }
+
+    @Test
+    fun `notifies lastNameRelay when lastName changed`() {
+        // GIVEN
+        initTestSubject()
+        val expected = "expected"
+
+        // WHEN
+        testSubject.onLastNameChanged(expected)
+
+        // THEN
+        verify(lastNameRelay).setValue(any(), any(), eq(expected))
+        verifyNoMoreInteractions(lastNameRelay)
+        verifyZeroInteractions(firstNameRelay, savedUserRelay)
+    }
+
+    @Test
+    fun `notifies savedUserRelay when SaveClicked`() {
```

Comment:
Oh neat.  That's actually awesome, and makes for much more readable tests.  I'm fine with converting them all to this (keeping some kind of consistent naming format like) and adding a line in the readme about these warnings and how to turn them off, as i think it is useful information in general.